Merge from CTF/shooter

rubenwardy opened the pull request:
This PR contains all fixes in CTF's fork of shooter. We've tried to keep the changes limited to things that upstream shooter may want to have. For all our changes, also look at the [shooter_tweaks](https://github.com/MT-CTF/capturetheflag/tree/master/mods/pvp/shooter_tweaks) mod.

You may not want to merge this all, but instead review and cherry-pick fixes. I'm also happy to make PRs for each fix if that would help you

* Re "Fix shooter.get_weapon_spec": I made a mistake when backporting get_weapon_spec to shooter originally. Here's the fix

rubenwardy committed:
Fix shooter.get_weapon_spec

rubenwardy committed:
Use correct puncher

ClobberXD committed:
Nerf shotgun

- Reduce magazine capacity
- Reduce fire-rate

rubenwardy commented:
This fix multiple things because I accidentally made it from master, I should split into multiple PRs

ClobberXD committed:
Crossbow: Pass player as puncher instead of arrow object

ClobberXD committed:
Fix inventory corruption when using automatic weapons

ClobberXD commented:
This PR now includes a fix for #51

ClobberXD commented:
There's no open issue that corresponds to https://github.com/stujones11/shooter/pull/48/commits/3d7ad1a6bc2a187366759fe1895a9afc1efc7641, but the commit passes the player object to `punch` instead of the arrow object - bc technically, it's the player "punching" the target, and not the arrow itself.

ClobberXD committed:
shooter: Fix invalid particlespawner definition

ClobberXD committed:
Ignore the shooter player when processing rounds (#1)

This removing the initial raycast offset and allows damaging players at point-blank range

LoneWolfHT committed:
Fix constant server crash

ClobberXD committed:
[trivial] api.lua: Fix code-style

ClobberXD committed:
Properly track players' LMB keypresses

Fixes automatic fire continuing after player's death and respawn

LoneWolfHT committed:
Add ranged damage group to guns and crossbow

* Add ranged damage group to guns

* Add ranged and crossbow group to crossbow

LoneWolfHT committed:
Fix infinite grapple hook uses

LoneWolfHT commented:
https://github.com/stujones11/shooter/pull/48/commits/d7f6dc7c06110ca7f1eead0b15b6630d7466c649 isn't exactly a bugfix, I can remove if wanted
